platform/darwin/src/MGLRasterSource.mm

```diff
@@ -6,18 +6,48 @@
 
 @implementation MGLRasterSource
 
-- (instancetype)initWithSourceIdentifier:(NSString *)sourceIdentifier URL:(NSURL *)url tileSize:(CGFloat)tileSize {
+- (instancetype)initWithSourceIdentifier:(NSString *)sourceIdentifier URL:(NSURL *)url tileSize:(CGFloat)tileSize
+{
     if (self = [super initWithSourceIdentifier:sourceIdentifier]) {
         _URL = url;
         _tileSize = tileSize;
     }
     return self;
 }
 
-- (std::unique_ptr<mbgl::style::Source>)mbglSource {
-    auto source = std::make_unique<mbgl::style::RasterSource>(self.sourceIdentifier.UTF8String,
-                                                              self.URL.absoluteString.UTF8String,
-                                                              uint16_t(self.tileSize));
+- (instancetype)initWithSourceIdentifier:(NSString *)sourceIdentifier tileURLTemplates:(NS_ARRAY_OF(NSString *) *)tileURLTemplates minimumZoomLevel:(NSUInteger)minimumZoomLevel maximumZoomLevel:(NSUInteger)maximumZoomLevel
+{
+    if (self = [super initWithSourceIdentifier:sourceIdentifier])
+    {
+        _tileURLTemplates = tileURLTemplates;
+        _minimumZoomLevel = minimumZoomLevel;
+        _maximumZoomLevel = maximumZoomLevel;
+    }
+    return self;
+}
+
+- (std::unique_ptr<mbgl::style::Source>)mbglSource
+{
+    std::unique_ptr<mbgl::style::RasterSource> source;
+    
+    if (self.URL)
+    {
+        source = std::make_unique<mbgl::style::RasterSource>(self.sourceIdentifier.UTF8String,
+                                                             self.URL.absoluteString.UTF8String,
```

Comment:
Call `-[NSURL(MGLAdditions) mgl_URLByStandardizingScheme]` in order to canonicalize asset: URLs and such.